zap/src/main/java/org/zaproxy/zap/scan/filters/GenericFilterBean.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2019 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.scan.filters;
+
+import java.util.Collection;
+import java.util.LinkedHashSet;
+
+/**
+ * @author KSASAN preetkaran20@gmail.com
+ *     <p>Generic Filter Bean which can be applied to any Class Object which implements hashCode and
+ *     equals
+ * @param <T>
+ */
+public class GenericFilterBean<T> {
```

Comment:
Can be used by Any Scan Filter. For now used by TagScanFilter, HttpStatusCodeScanFilter and HttpMethodScanFIlter

Reply:
That will address this: https://github.com/zaproxy/zaproxy/issues/2819 (At least partially)